embedded/tbtree/tbtree.go

```diff
@@ -2385,13 +2405,19 @@ func (l *leafNode) split() (node, error) {
 	newLeaf.updateTs()
 
 	l.values = l.values[:splitIndex]
-
 	l.updateTs()
 
-	metricsBtreeLeafNodeEntries.WithLabelValues(l.t.path).Observe(float64(len(l.values)))
-	metricsBtreeLeafNodeEntries.WithLabelValues(l.t.path).Observe(float64(len(newLeaf.values)))
+	ns1, err := l.split()
```

Comment:
How can we be sure that this and the `newLeaf.split()` below won't end up with an infinite recursive split?

I'm also worried about this ending up with very uneven split - like if the first half must be split again, it will end up with 3 children, first two containing ~1/4 entries, the last one containing 1/2.

Reply:
termination is guaranteed by ensuring at least two entries can fit into a inner node

Reply:
What about the uneven distribution of child nodes in case any of new nodes needs a recursive split? Maybe better to base the split position based on bytes rather than entries?

Reply:
As discussed offline - those recursive splits will be very rare and will only happen in a very odd key length distributions (such as majority of small keys and few huge ones). It should be considered a safely recovery (and thus an unusual situation) rather than a common, optimizable scenario.